### PR TITLE
storage: better logging on replica clobbering

### PR DESCRIPTION
**Pull request by tbg**
See #7224.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/7227)

### CODE REVIEW

**Comment by mberhault** [comment]
LGTM

File: storage/store.go [hunk 1]
```diff
@@ -2237,14 +2237,19 @@ func (s *Store) replicaDescriptorLocked(groupID roachpb.RangeID, replicaID roach
...
-		if old != replica {
-			log.Fatalf("%s replicaDescCache: clobbering %s with %s", s, old, replica)
+func (s *Store) cacheReplicaDescriptorLocked(
+	groupID roachpb.RangeID, rDesc roachpb.ReplicaDescriptor,
+) {
+	if old, ok := s.mu.replicaDescCache.Get(replicaDescCacheKey{groupID, rDesc.ReplicaID}); ok {
+		if old != rDesc {
+			rpl, _ := s.getReplicaLocked(groupID)
+			log.Fatalf("store %d, range %d: clobbering %+v with %+v "+
+				"in replicaDescCache; have replica %+v", groupID, s.Ident.StoreID,
```
Comment (mberhault) [inline storage/store.go]: looks like you're printing "store: <groupID>, range: <storeID>" etc...

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/store.go, line 2241 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227#-KKFmvsTkcKiYvqhONVR:-KKFmvsUuAjAnKQwADxZ:-780672552) ([raw file](https://github.com/cockroachdb/cockroach/blob/1fbda89f337113f3b87e3c2280434d9ca0cdf31f/storage/store.go#L2241)):*
> ```Go
> // cacheReplicaDescriptorLocked requires that the store lock is held.
> func (s *Store) cacheReplicaDescriptorLocked(
> 	groupID roachpb.RangeID, rDesc roachpb.ReplicaDescriptor,
> ```

nit: replicaDesc? `rDesc` is really not any less ambiguous

---

*[storage/store.go, line 2247 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227#-KKFi3KykpAs5KXorIji-r1-2247:-KKFn78ZSIbXoUPglvZT:-1718311555) ([raw file](https://github.com/cockroachdb/cockroach/blob/1fbda89f337113f3b87e3c2280434d9ca0cdf31f/storage/store.go#L2247)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary>
> looks like you're printing `store: <groupID>, range: <storeID>` etc...
</details>
might also just print all of `s.Ident`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[storage/store.go, line 2245 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227#-KKFnhWODhJA8VdSVLJD:-KKFnhWODhJA8VdSVLJE:198002027) ([raw file](https://github.com/cockroachdb/cockroach/blob/739adaab684cbfbb132573308927965c95c15269/storage/store.go#L2245)):*
> ```Go
> 	if old, ok := s.mu.replicaDescCache.Get(replicaDescCacheKey{groupID, rDesc.ReplicaID}); ok {
> 		if old != rDesc {
> 			rpl, _ := s.getReplicaLocked(groupID)
> ```

logging this doesn't look super useful, but maybe that's my untrained eye

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227)*

**Comment by tschottdorf** [comment]
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[storage/store.go, line 2241 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227#-KKFmvsTkcKiYvqhONVR:-KKFoIY7ZKJ5S4LM5SXp:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/1fbda89f337113f3b87e3c2280434d9ca0cdf31f/storage/store.go#L2241)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> nit: replicaDesc? `rDesc` is really not any less ambiguous
</details>
Done.

---

*[storage/store.go, line 2247 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227#-KKFi3KykpAs5KXorIji-r1-2247:-KKFoI3LuMFfWR42NcqZ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/1fbda89f337113f3b87e3c2280434d9ca0cdf31f/storage/store.go#L2247)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> might also just print all of `s.Ident`
</details>
Done.

---

*[storage/store.go, line 2245 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227#-KKFnhWODhJA8VdSVLJD:-KKFoErMVi_JD-C50Pg8:828883950) ([raw file](https://github.com/cockroachdb/cockroach/blob/739adaab684cbfbb132573308927965c95c15269/storage/store.go#L2245)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> logging this doesn't look super useful, but maybe that's my untrained eye
</details>
Yeah, let's find out whether it is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/7227)*

**Pull request merged**